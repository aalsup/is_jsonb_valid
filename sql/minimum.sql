-- minimum validation
-- above the minimum is valid
SELECT is_jsonb_valid('{"minimum":1.1}', '2.6');
-- boundary point is valid
SELECT is_jsonb_valid('{"minimum":1.1}', '1.1');
-- below the minimum is invalid
SELECT is_jsonb_valid('{"minimum":1.1}', '0.6');
-- ignores non-numbers
SELECT is_jsonb_valid('{"minimum":1.1}', '"x"');
-- exclusiveMinimum validation
-- above the minimum is still valid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":true}', '1.2');
-- boundary point is invalid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":true}', '1.1');