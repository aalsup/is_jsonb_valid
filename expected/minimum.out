-- minimum validation
-- above the minimum is valid
SELECT is_jsonb_valid('{"minimum":1.1}', '2.6');
 is_jsonb_valid 
----------------
 t
(1 row)

-- boundary point is valid
SELECT is_jsonb_valid('{"minimum":1.1}', '1.1');
 is_jsonb_valid 
----------------
 t
(1 row)

-- below the minimum is invalid
SELECT is_jsonb_valid('{"minimum":1.1}', '0.6');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores non-numbers
SELECT is_jsonb_valid('{"minimum":1.1}', '"x"');
 is_jsonb_valid 
----------------
 t
(1 row)

-- exclusiveMinimum validation
-- above the minimum is still valid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":true}', '1.2');
 is_jsonb_valid 
----------------
 t
(1 row)

-- boundary point is invalid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":true}', '1.1');
 is_jsonb_valid 
----------------
 f
(1 row)

